### DISCUSSION TITLE
Provide nice API to unwrap value you get back from action

### DISCUSSION THREAD

**Discussion by nikivdev** [next-safe-action/next-safe-action#132] [Ideas]
<img width="546" alt="image" src="https://github.com/TheEdoRan/next-safe-action/assets/6391776/a21e3812-8583-421f-b2ec-7b1ba8fe0b5e">

```typescript
const profileLoadMostPostsSchema = z.object({
  username: z.string(),
  pageNumber: z.number(),
})

export const profileLoadMostPostsAction = actionClient
  .schema(profileLoadMostPostsSchema)
  .action(async ({ parsedInput: { username, pageNumber } }) => {
    const session = auth.getSession()
    const client = session.client
    try {
      const res = await profileLoadMorePosts.run(client, {
        username,
        pageNumber,
      })
      return res
    } catch {
      return { failure: "Error with EdgeDB" }
    }
  })
```
	
Have action. How do I unwrap nicely, can you provide a nice API for this please. 🙏
	
```typescript
const posts = await profileLoadMostPostsAction({
  username: "nikiv",
  pageNumber: local.pageNumber.get(),
})
```

**Comment by nikivdev** [comment]
<img width="763" alt="image" src="https://github.com/TheEdoRan/next-safe-action/assets/6391776/dd1a34c7-780b-4a5d-92e3-643bd921a921">

This is wrong.

<img width="439" alt="image" src="https://github.com/TheEdoRan/next-safe-action/assets/6391776/1a31df0a-4152-4f81-9867-135c2bc3c126">


this is me logging `posts.data`

**Comment by nikivdev** [comment]
This works:

```typescript
const posts = await profileLoadMostPostsAction({
  username: "nikiv",
  pageNumber: local.pageNumber.get(),
})

authData.createdPosts.set([
  ...(authData.createdPosts.get() ?? []),
  ...posts.data[0].createdPosts,
])
```

I wonder why it has to be array though.

**Reply by TheEdoRan** [reply]
Sorry but I really don't understand what's happening here. `action.data` has the same type of `serverCodeFn`'s return, so in this case it seems to be  `{ createdPosts: post[] }[] | { failure: string }`. However I'm not familiar with EdgeDB. Can you please provide a minimal reproduction example link? Thanks.

**Reply by nikivdev** [reply]
Yea you are right, fixed it, just forgot how to unwrap the value.
